functions.py

```diff
@@ -49,12 +49,10 @@ def sama_counter(week):
 
 def attendance_counter(first_week, last_week):
     from inputs import raw_result_array
-    week_index = week - 17
-    rss_per_user_week_dict = get_and_load_dictionary(raw_result_array[week_index])
-
+    all_weeks_rss_data = asyncio.run(batch_fetch(raw_result_array[first_week-1:last_week+1])) # fetch data for all specified weeks
```

Comment:
since attendance is based on multiple weeks and raw_result_array is an array of urls for api get requests representing each carnage week, we can batch them and run them asynchronously. note, we only do this data fetch once.

Reply:
also, just something to think about...some players have had multiple user names over the course of the weeks. not really much can be done about connecting the data unless someone stepped forward and confirmed all of their past usernames.

another thing...if we calculate attendance today and then want to recalculate two weeks later, we effectively recompute for all past weeks too. might be fun to play around with caching. 

mightconsider making a small database to store all the carnage week data. we call the api everytime we call a function, get mostly the same data, and do stuff with it. idea is to store carnage data and only call mcapi api when we need data that we do not already have stored (reducing number of api calls to their endpoints). 

kinda overkill because we are not even nearly exhausting their api (given these functions are not ran super often), but cool to play with.